packages/data-context/src/data/ProjectConfigManager.ts

```diff
@@ -461,6 +461,15 @@ export class ProjectConfigManager {
     )
   }
 
+  private getRepoRoot = () => {
+    /*
+      repoRoot is used to detect the full file path for the relativeFile in stack_utils
+      it's needed to show the correct link to files in repo mgmt tools like GitHub in the dashboard
+      right now we assume it's same as `.git` dir, if need be can update to look for most root-level package.json
+    */
+    return this.options.ctx.git?.gitBaseDir
```

Comment:
Will this work for projects that use git submodules?

Reply:
Oh dear, I hadn't heard of git submodules. From what I can see, the previous implementation did not support git submodules and these changes do not either. Additionally, the top results on Google regarding git submodules are from ~2015/2016, and a 2016 blog post from GitHub says:
> Before you add a repository as a submodule, first check to see if you have a better alternative available. Git submodules work well enough for simple cases, but these days there are often better tools available for managing dependencies than what Git submodules can offer.

I think we might be able to get some hard numbers from the data team on their usage by paid users, but the old results + recommendation from GitHub to use alternatives makes me think the use of git submodules is likely not very common.

Given the existing lack of git submodule support and likeliness of this being a rare edge case, I would consider adding support a separate issue and would recommend fixing it if/when we run into it. What do you think?

Reply:
@rachelruderman, for the incorrect links where you were seeing an ```@``` included in the URL, did you ever determine the cause? Submodules are something that I had briefly considered being related to that. The [original Jira](https://cypress-io.atlassian.net/browse/CLOUD-599) logged on the Cloud side has a specific link for a client with an ```@``` in the URL that may still be valid. It may require that you impersonate someone with access to that project in the dashboard to see it, just fyi.

A [quick search](https://sourcegraph.com/search?q=context:global+file:.gitmodules&patternType=standard) finds some big projects that are using submodules, so we may not want to completely rule it out as something to address. However, I would definitely support the creation of a separate issue to track that work. Before picking it up, we could sync with a data scientist to help us understand how impactful a fix there may be and then use that info to prioritize the change.

Reply:
Ahhh I forgot about the `@` -- it was called out in the [original JIRA](https://cypress-io.atlassian.net/browse/CLOUD-599), but the "Good" / "Bad" examples in the [GitHub issue](https://github.com/cypress-io/cypress/issues/22971) were just for monorepos 🤦‍♀️ 

AFAIK this PR would not address the `@` issue. Does it make sense to create a separate GitHub issue for that? Or would you like me to continue with this ticket in the upcoming sprint and incorporate it?

Reply:
My thoughts:
* Let's figure out what the situation with the ```@``` is all about (is that for submodules or something else?)
* Then let's verify whether this fix resolves that situation too
* Once we understand that, we can determine if we want to split it out to another story or continue working on it from this one

Most likely, we'll want to split it out, given where we're at on this sprint. Unless the situation with the ```@``` is a pretty common one (and it impacts a lot of people), the priority of the remaining work may be lower than what you have here. And in that case, pushing these current changes out will at least resolve the issue for users with monorepos.

Reply:
Re: git submodules, i created a [ticket for the data team](https://cypress.clickup.com/t/18033298/DATA-904) and they got back to us. It seems they might not have enough information on their end to detect git submodules